### ISSUE TITLE
test 13 - stack fault

### ISSUE THREAD

**Issue by philiprbrenan** [philiprbrenan/NasmX86#11]
@majorendian  Here is my version of your very useful test (also attached as a zip). It does not cause a seg fault: if you are willing, please look further for the cause of the problem, which I think, must occur in your test code not x86.pm?
```
#Testing of local data variables on the stack
# Tino 2021/05/14
use Test::Most tests => 1;
use Nasm::X86 qw(:all);

if (1) {                                                                        #TLocalData #TLocalData::variable #TLocalData::free #TLocalData::start
  my $vars = LocalData;
  my $a = $vars->variable( 8,'a');                                              # Create an 8 byte variable 'a'
  my $b = $vars->variable(16,'b');                                              # Create a 16 byte variable 'b'
  $vars->start;                                                                 # Allocate space on the stack for the variables
  Mov 'qword'.$a->stack, 10;                                                    # Initialize a
  Mov 'qword'.$b->stack,  4;                                                    # Initialize b
  Mov rax, $a->stack;                                                           # Get a
  Add rax, $b->stack;                                                           # Add b
  $vars->free;                                                                  # Free stack
  PrintOutRegisterInHex rax;                                                    # Print result

  is_deeply Assemble, <<END;                                                    # Assemble and run
   rax: 0000 0000 0000 000E
END
```
[13.zip](https://github.com/philiprbrenan/NasmX86/files/6481364/13.zip)

**Comment by majorendian** [comment]
Your version leaves out a CALL instruction. The issue was mainly about return-address corruption on the stack via a MOV instruction into the address pointed to by RBP+8. I have left test 13 as you have made it with some minor changes to check for other things and moved the return address segfault test to test 14.

**Comment by philiprbrenan** [comment]
I appreciate the clarification.  I think your plan to focus on just the Call instruction is the right way to go.

**Comment by philiprbrenan** [comment]
Resolved in #15 t
